.github/workflows/ci-core-reusable.yml

```diff
@@ -442,11 +450,27 @@ jobs:
         run: |
           ci_run ./bin/run_on_all_chains.sh "zkstack dev test integration --no-deps --ignore-prerequisites --external-node" ${{ env.CHAINS }} ${{ env.INTEGRATION_TESTS_LOGS_DIR }}
 
+      - name: Repeat integration tests en on push to main to check for flakiness
+        if: ${{ (steps.condition.outputs.should_run == 'true' && github.event_name == 'push' && github.ref == 'refs/heads/main') }}
+        run: |
+          for i in {1..10}; do
+            echo "Iteration $i"
+            ci_run ./bin/run_on_all_chains.sh "zkstack dev test integration --no-deps --ignore-prerequisites --external-node" ${{ env.CHAINS }} ${{ env.INTEGRATION_TESTS_LOGS_DIR }}
+          done
+
       - name: Fee projection tests
         run: |
           ci_run killall -INT zksync_server || true
           ci_run ./bin/run_on_all_chains.sh "zkstack dev test fees --no-deps --no-kill" ${{ env.CHAINS }} ${{ env.FEES_LOGS_DIR }}
 
+      - name: Repeat fee projection tests on push to main to check for flakiness
+        if: ${{ (steps.condition.outputs.should_run == 'true' && github.event_name == 'push' && github.ref == 'refs/heads/main') }}
+        run: |
+          for i in {1..10}; do
+            echo "Iteration $i"
+            ci_run ./bin/run_on_all_chains.sh "zkstack dev test fees --no-deps --no-kill" ${{ env.CHAINS }} ${{ env.FEES_LOGS_DIR }}
```

Comment:
Probably not strictly necessary but this is what I did to make sure all logs are saved: [`da4dd2f` (#3504)](https://github.com/matter-labs/zksync-era/pull/3504/commits/da4dd2fd3ea7b7e41c29ba979790ceb440d3cc54#diff-293b35dbbaf364cec87320ca4091f93d974165e35cc640d7f44cfb9f51ebed0dR521-R523)

Reply:
Also, making sure server logs are saved is a bit more involved but also possible

Reply:
Makes sense, corrected